kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/PreLinkExtension.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.konan
+
+import org.jetbrains.kotlin.backend.common.CommonBackendContext
+import org.jetbrains.kotlin.backend.konan.objcexport.ObjCExportNamer
+import org.jetbrains.kotlin.extensions.ProjectExtensionDescriptor
+
+interface PreLinkExtension {
+    companion object :
+        ProjectExtensionDescriptor<PreLinkExtension>(
+            "org.jetbrains.kotlin.preLinkExtension", PreLinkExtension::class.java
+        )
+
+    fun process(context: PreLinkContext): PreLinkResult?
+}
+
+data class PreLinkResult(
+    val additionalObjectFiles: List<ObjectFile>,
+    val additionalLinkerFlags: List<String>,
+)
+
+interface PreLinkContext {
+    val config: KonanConfig
+    val backendContext: CommonBackendContext
```

Comment:
Both `KonanConfig` and `CommonBackendContext` are poorly designed classes, especially for this purpose.
What exactly do you use in your plugin from these classes? Is it feasible to include only those parts here?